hs/src/Reach/AST/SL.hs

```diff
@@ -160,6 +160,201 @@ data SLVal
   | SLV_Deprecated Deprecation SLVal
   deriving (Eq, Generic)
 
+class NewerVar a where
+  containsVarNewerThan :: Int -> a -> Bool
+
+instance (NewerVar a) => NewerVar [a] where
+  containsVarNewerThan ret = all (containsVarNewerThan ret)
+
+instance (NewerVar b) => NewerVar (a,b) where
```

Comment:
This feels dangerous... because there are some structures in Reach that have real information on both sides, I think that the client (`Struct`?) should use `map snd`